Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -629,6 +629,10 @@ FontCascade::CodePath FontCascade::codePath(const TextRun& run, std::optional<un
     if (s_codePath != CodePath::Auto)
         return s_codePath;
 
+    // FIXME: add support for text-autospace on simple path (rdar://133319627).
+    if (textAutospace().hasIdeographAlpha())
+        return CodePath::Complex;
+
```

Comment:
One thing we could do is, while we're scanning the text for a complex character, also scan it for ideographs -- if there are none, then we can continue on the simple path.

Reply:
Good idea